app/helpers/fhir_helper.rb

```diff
@@ -0,0 +1,329 @@
+# frozen_string_literal: true
+
+# Helper module for FHIR translations
+module FhirHelper # rubocop:todo Metrics/ModuleLength
+  # Returns a representative FHIR::Patient for an instance of a Sara Alert Patient. Uses US Core
+  # extensions for sex, race, and ethnicity.
+  # https://www.hl7.org/fhir/us/core/StructureDefinition-us-core-patient.html
+  def patient_as_fhir(patient)
+    creator_agent_ref = FHIR::Reference.new
+    creator_app = OauthApplication.where(user_id: patient.creator.id).first
+    if creator_app
+      # Created with an M2M workflow client
```

Comment:
This is not an accurate check. Every API user/client, regardless of workflow, as an associated OAuth Application. So `creator_app` could very well be true if someone is using the User workflow and happens to be working with an OAuth app that is also registered for the M2M workflow (and therefore has a `user_id` associated with it). I think you either a) always want to just show a user ID for the "creator" in these cases, because in the case of the M2M workflow you can use the API proxy user ID instead of the app name or b) check to see if the user if the `patient.creator.id` is an API proxy user (`is_api_proxy`) and that will tell you.

Reply:
Good catch. I would like to return the app name since that will likely be more meaningful for a human debugging a client than the user ID, so I think I'll go with your b suggestion.

Reply:
Fixed here https://github.com/SaraAlert/SaraAlert/pull/521/commits/3258b7a3f622672dcbab96490cb7fc30c257c27f